cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -1629,11 +1629,23 @@
 
 ;; Put a value into a GPR.
 ;;
-;; Asserts that the value goes into a GPR.
+;; Moves the value into a GPR if it is a type that would naturally go into an
+;; XMM register.
 (decl put_in_gpr (Value) Gpr)
+
+;; Case for when the value naturally lives in a GPR.
 (rule (put_in_gpr val)
+      (if-let (value_type ty) val)
+      (if-let (type_register_class (RegisterClass.Gpr $true)) ty)
       (gpr_new (put_in_reg val)))
```

Comment:
Should this match against `(RegisterClass.Gpr _)` instead, to avoid potentially backtracking here instead of forcing the assertion failure in `put_in_reg` like before?

Reply:
Good idea.